go-controller/pkg/factory/factory.go

```diff
@@ -439,7 +543,48 @@ func (wf *WatchFactory) GetNamespaces() ([]*kapi.Namespace, error) {
 	return namespaceLister.List(labels.Everything())
 }
 
-// GetFactory returns the underlying informer factory
-func (wf *WatchFactory) GetFactory() informerfactory.SharedInformerFactory {
-	return wf.iFactory
+func (wf *WatchFactory) NodeInformer() cache.SharedIndexInformer {
+	return wf.informers[nodeType].inf
+}
+
+// LocalPodInformer returns a shared Informer that may or may not only
+// return pods running on the local node.
+func (wf *WatchFactory) LocalPodInformer() cache.SharedIndexInformer {
+	return wf.informers[podType].inf
+}
+
+func (wf *WatchFactory) PodInformer() cache.SharedIndexInformer {
+	return wf.informers[podType].inf
+}
+
+func (wf *WatchFactory) NamespaceInformer() cache.SharedIndexInformer {
+	return wf.informers[namespaceType].inf
+}
+
+// serviceLabelSelector returns a label selector that excludes headless
+// services and services with an alternate proxy selected.
+// Used for Services and Endpoints.
+//
+// Matches the value upstream in cmd/kube-proxy/app/server.go
+func serviceLabelSelector() func(options *metav1.ListOptions) {
+	// if the proxy-name annotation is set, skip this service
+	noProxyName, err := labels.NewRequirement("service.kubernetes.io/service-proxy-name", selection.DoesNotExist, nil)
+	if err != nil {
+		// cannot occur
+		panic(err)
+	}
+
+	// if the service is headless, skip it
```

Comment:
nit, this is for skipping endpoints

Reply:
well, services are headless, not endpoints :-)  (tail-less?)

Reply:
My reading is that the service does not have the label IsHeadlless, the endpoints does, is the endpoint controller who adds it, 
https://github.com/kubernetes/kubernetes/blob/104ad794e5d42cd5c4317bd547d85db1dc2c31fa/pkg/controller/endpoint/endpoints_controller.go#L508-L512
so you filter the endpoints events only, for the service you still have to check that ClusterIP=None, there is no filtering
Anyway, this is totally a nitpick but an useful discussion to me to review this logic  :smile:

Reply:
you are right; fixed.